src/main/kotlin/org/rust/lang/core/macros/RsExpandedElement.kt

```diff
@@ -233,6 +239,48 @@ private fun Int.fromBodyRelativeOffset(call: RsMacroCall): Int? {
     return elementOffset
 }
 
+private fun MappedTextRange.fromBodyRelativeRange(call: RsMacroCall): MappedTextRange? {
+    val newSrcOffset = srcOffset.fromBodyRelativeOffset(call) ?: return null
+    return MappedTextRange(newSrcOffset, dstOffset, length)
+}
+
+fun RsMacroCall.mapRangeFromExpansionToCallBodyStrict(range: TextRange): TextRange? {
+    return mapRangeFromExpansionToCallBody(range).singleOrNull()?.takeIf { it.length == range.length }
+}
+
+private fun RsMacroCall.mapRangeFromExpansionToCallBody(range: TextRange): List<TextRange> {
+    val expansion = expansion ?: return emptyList()
+    return mapRangeFromExpansionToCallBody(expansion, this, range)
+}
+
+private fun mapRangeFromExpansionToCallBody(
```

Comment:
Do we really need all `mapRangeFromExpansionToCallBody*` functions?
It seems we need only `mapRangeFromExpansionToCallBodyStrict` because it public and `mapRangeFromExpansionToCallBody(org.rust.lang.core.macros.MacroExpansion, org.rust.lang.core.psi.RsMacroCall, org.rust.lang.core.macros.MappedTextRange)` because it's recursive.

Maybe it's better to inline other ones? They look quite small and add only additional overloads of `mapRangeFromExpansionToCallBody` that can be confusing. Also, such inlining allows us to rename `mapRangeFromExpansionToCallBodyStrict` to `mapRangeFromExpansionToCallBody` - looks like `strict` doesn't add additional info because we don't have public non strict version

Reply:
Non-strict version is needed for the (ongoing) highlighting. Another one overload replaces `MappedTextRange` to `TextRange`.